mutation/async_utils.hh

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "mutation_partition.hh"
+#include "mutation.hh"
+
+//
+// Applies p to current object.
```

Comment:
s/current/target/

Reply:
Right.  copy-pasted from the respective `apply` method.  Will fix.